### ISSUE TITLE
JSON.stringify(arrayOfParseObjects) throws TypeError: seen.indexOf is not a function

### ISSUE THREAD

**Issue by superbigsoft** [parse-community/Parse-SDK-JS#435]
I'm having _TypeError: seen.indexOf is not a function_ in an CordovaApp running on **iOS 9.3**. (The error does not happen on Chrome/Safari or iOS 10)
=> Please let me know if the work-around below does not cause any side effect. Thanks all.

I have following Parse Object structures:
+ User : 1 entry
+ House { owner: Pointer<User> }: 3 entries with owner pointed the only User.

Then following code would throw the error:

```
var query = Parse.Query("House");
query.find().then(function(houses) {
    JSON.stringify(houses); // This would throw the following error with provided stack
})

```

> TypeError: seen.indexOf is not a function. (In 'seen.indexOf(seenEntry)', 'seen.indexOf' is undefined)
> encode — parse.js:11238
> default — parse.js:11190
> value — parse.js:4286
> stringify

**Work-around:** I added a check to make sure seen is an array.

> key: 'toJSON',
>           value: function (seen) {
>             var seenEntry = this.id ? this.className + ':' + this.id : this;
>             **var seen = Array.isArray(seen)? seen: [seenEntry];**
>             var json = {};
>             var attrs = this.attributes;
>

**Comment by pewh** [comment]
I also encounter same issue. Your workaround is working. Somebody please make an PR. It's frustated to always edit this file whenever execute `yarn` or `npm`.

**Comment by flovilmart** [comment]
@pewn could you make a Pr and a test?

**Comment by felipemartim** [comment]
Can confirm this fixes the issue on iOS 9.3.
